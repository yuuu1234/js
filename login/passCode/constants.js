export const PASSCODE_MAX_LENGTH = 6;

export default {
  PASSCODE_MAX_LENGTH,
};
