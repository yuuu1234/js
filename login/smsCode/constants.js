import numeral from 'numeral';

export const COUNTDOWN_TEXT = 'Code Expires in';
export const UNITTEXT = 'secs';
export const SECONDS_LIMIT = 60;
export const NUMBER_OF_DIGITS = 6;
export const TITLE_USER_SMS_INPUT_SCREEN = 'We Are Sending You An SMS Code';
export const FIREBASE_ERROR_MESSAGES = {
  'auth/invalid-phone-number': 'The phone number is too long.',
  'auth/invalid-verification-code': 'The SMS code is invalid. Please double check or resend it.',
  default: 'There has been an error processing your request.',
};

export function isSmsCodeValid(code, length) {
  return code && code.length === length && code === numeral(code).value().toString();
}
