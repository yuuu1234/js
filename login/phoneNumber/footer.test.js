import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import Footer from './footer';
import * as navigatorConstants from '../../../constants/navigatorConstants';

describe('Footer', () => {
  const props = {
    loginStore: {
      login: { isPhoneNumberVerified: true },
      isLoading: false,
    },
    handleNavigation: jest.fn(),
  };
  const component = <Footer {...props} />;
  let wrapper;

  beforeEach(() => {
    wrapper = mount(component);
  });

  it('should render component correctly', () => {
    const tree = renderer.create(component).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('should call navigate with PasscodeLogin if user press FormButton', () => {
    const formButton = wrapper.find('FormButton');
    formButton.simulate('click');

    expect(props.handleNavigation).toHaveBeenCalledWith(navigatorConstants.PasscodeLogin);
  });

  it('should set LoadingModal isShow to true if isLoading is true', () => {
    let loadingModal = wrapper.find('LoadingModalComponent');
    expect(loadingModal.props().isShow).toBeFalsy();

    wrapper.setProps({ loginStore: { login: { isPhoneNumberVerified: true }, isLoading: true } });
    wrapper.update();
    loadingModal = wrapper.find('LoadingModalComponent');

    expect(loadingModal.props().isShow).toBeTruthy();
  });

  it('should disable FormButton if isPhoneNumberVerified is false', () => {
    let formButton = wrapper.find('FormButton');
    expect(formButton.props().disable).toBeFalsy();

    wrapper.setProps({ loginStore: { login: { isPhoneNumberVerified: false } } });
    wrapper.update();
    formButton = wrapper.find('FormButton');

    expect(formButton.props().disable).toBeTruthy();
  });
});
