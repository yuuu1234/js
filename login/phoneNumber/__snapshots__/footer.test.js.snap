// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render component correctly 1`] = `
<View
  onLayout={[Function]}
  style={
    Object {
      "paddingBottom": 0,
    }
  }
>
  <View
    style={
      Object {
        "marginBottom": 25,
        "marginTop": 10,
        "width": "100%",
      }
    }
  >
    Object {
      "children": Array [
        <Text
          style={
            Object {
              "color": "#fff",
              "fontSize": 16,
              "fontWeight": "bold",
            }
          }
        >
          Continue
        </Text>,
      ],
      "props": Object {
        "accessibilityLabel": "FormButton",
        "accessible": true,
        "focusable": true,
        "onClick": [Function],
        "onResponderGrant": [Function],
        "onResponderMove": [Function],
        "onResponderRelease": [Function],
        "onResponderTerminate": [Function],
        "onResponderTerminationRequest": [Function],
        "onStartShouldSetResponder": [Function],
        "style": Array [
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "#003D9A",
            "borderRadius": 15,
            "height": 41,
            "justifyContent": "center",
            "marginBottom": 16,
            "marginHorizontal": 49,
            "marginTop": 6,
          },
          false,
          Object {},
        ],
      },
      "type": "View",
    }
    <Text
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "color": "#929292",
          "fontSize": 16,
          "justifyContent": "center",
          "textDecorationLine": "underline",
        }
      }
    >
      Changed your number?
    </Text>
    <Modal
      hardwareAccelerated={false}
      onDismiss={[Function]}
      transparent={true}
      visible={false}
    />
  </View>
</View>
`;
